Doc/c-api/hash.rst

```diff
@@ -41,8 +45,33 @@ See also the :c:member:`PyTypeObject.tp_hash` member.
    .. versionadded:: 3.4
 
 
+Functions
+^^^^^^^^^
+
+.. c:function:: int Py_HashDouble(double value, Py_hash_t *result)
+
+   Hash a C double number.
+
+   * Set *\*result* to the hash and return ``1`` if *value* is finite or is
+     infinity.
+   * Set *\*result* to :data:`sys.hash_info.nan <sys.hash_info>` (``0``) and
+     return ``0`` if *value* is not-a-number (NaN).
+
+   *result* must not be ``NULL``.
+
+   .. note::
+      Only rely on the function return value to distinguish the "not-a-number"
+      case. *\*result* can be ``0`` if *value* is finite. For example,
+      ``Py_HashDouble(0.0, &result)`` sets *\*result* to 0.
+
+   .. versionadded:: 3.13
+
+
 .. c:function:: PyHash_FuncDef* PyHash_GetFuncDef(void)
 
    Get the hash function definition.
 
+   .. seealso::
```

Comment:
Nit: this change is unrelated to the addition of `Py_HashDouble`, right?

Reply:
Right, I reverted it. I wrote https://github.com/python/cpython/pull/112647 for this doc change.